### ISSUE TITLE
ExoPlayer fails to prepare for UHD MPEG-TS stream (worked before)

### ISSUE THREAD

**Issue by stari4ek** [google/ExoPlayer#7177]
### [REQUIRED] Issue description
With changes made in 6946170d3ece533655a23f8fe65936412f365a6c ExoPlayer fails to prepare provided mpeg-ts streams.
Works well with bd8ee155af5c441fa2dd0440f94638853b9ac7ee (-1 revision).
No issues in mplayer/vlc.

There is warning in the log, but it's logged in working/not-working cases:
```
W/PesReader: Unexpected start code prefix: 2217538
```

### [REQUIRED] Reproduction steps
Play provided streams in demo player.

### [REQUIRED] Link to test content
* https://drive.google.com/open?id=1JDmX-so5INSdAWGmZsRowACW3ZnWNkI5

There're 3 video streamed from same device (DREAMBOX 8000 HD). "work_01" works on any version of ExoPlayer. "notwork_*" - only with ExoPlayer using bd8ee155af5c441fa2dd0440f94638853b9ac7ee

### [REQUIRED] A full bug report captured from the device
not device-specific. Reproduced on nvidia shield/Sony TVs

### [REQUIRED] Version of ExoPlayer being used
git revisions are specified above

### [REQUIRED] Device(s) and version(s) of Android being used
not device-specific. Reproduced on nvidia shield/Sony TVs

**Comment by stari4ek** [comment]
Reverting next changes (https://github.com/google/ExoPlayer/commit/6946170d3ece533655a23f8fe65936412f365a6c#diff-11d839d2e0e7ccd61baa89293ddbd8f0L581):
```diff
-        if (streamType == 0x06) {
+        if (streamType == 0x06 || streamType == 0x05) {
```
make streams readable

**Comment by stari4ek** [comment]
It looks like there is `AIT` declared in PMT with 2821 pid, but there is no transport packets for it. (checked with `notwork_01`)

So, `maybeFinishPrepare` is never satisfied, due missing format for metadata tracks sample queue.

BTW, is there any reason to **require** `METADATA` (!) samples/track to be fully mpeg-ts complaint? (I do not have any expertise here. But is it valid case, when there is header/program data only?)

**Comment by stari4ek** [comment]
I'm thinking about next workarounds:
* flag to ignore AIT (like `FLAG_IGNORE_SPLICE_INFO_STREAM`)
* `PassthroughSectionPayloadReader.init` setting `Format`. If it can be dynamic - set initial format and update with offset later.

**Comment by andrewlewis** [comment]
@icbaker, @AquilesCanta Please could you take a look? This seems to be related to #6922.

**Comment by icbaker** [comment]
Thanks for the report - I can reproduce `notwork_01.ts` playing just before the DVB AIT change, and failing to play after it.

I'll take a deeper look on Monday.

**Comment by icbaker** [comment]
I think the underlying root cause for this is actually https://github.com/google/ExoPlayer/commit/94315ab757505f98d0b907fa2f7d2c516356ecae where we removed the format-setting from `SpliceInfoSectionReader#init()` which then became `PassthroughSectionReader`. Putting back that `format()` call back in makes `notwork_01.ts` and `notwork_02.ts` play for me.

However it makes the tests fail (which is why I removed it in the first place) :/

Seems the tests are over-strict - I'll work on relaxing the assertions.
